.github/workflows/cmake.yml

```diff
@@ -0,0 +1,71 @@
+name: CMake CI
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  build-and-test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        qt_version: [5, 6]
+        python_version: ["3.8", "3.9", "3.10"]
+      fail-fast: false
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python_version }}
+
```

Comment:
Noticed that you missed the apostrophes here, you need
```yaml
with:
  python-version: '${{ matrix.python-version }}'
```
otherwise it won't work.
It would really be much easier for you to just take the setup part form `build_latest.yaml`, which is working, instead of trying to do it all manually again.
In case you haven't given up on this...